### ISSUE TITLE
List of variables in statistics

### ISSUE THREAD

**Issue by FASTNAVi** [NPBruce/valkyrie#952]
Hi,

I really like the possibility to see used events in stats! I would like to see the same thing with variables at the moment of the end of scenario. That can help to see what exactly was used or which setup was used (if you are using vars for using different tiles or events..).

Thanks in advance for implementing this into Valkyrie ;-)

**Comment by scrubbless** [comment]
Would it be possible to have a variable type that could be output to the editor.

Either a new car type (% as an example) which pmis purely used to be output into the reviews sheet?

For example #output. 

I've been considering adding a score system to my scenarios and have it display at the end. Having it also printed to the reviews would be great for stats and reporting.

**Comment by BenRQ** [comment]
Why not just display all vars ? Do you see an issue with that ?
It's actually easier on code side to list all vars rather than just some of them.

**Comment by scrubbless** [comment]
Mainly about the amount of cells it will take up in the excel. I sure don't want all of Strain on Reality bars output. There are hundreds (maybe thousands)

**Comment by FASTNAVi** [comment]
I agree with BenRQ. In one cell can be a lot of variables (from the moment of the end of the game). You can copy them outside and then use search in text editor to find what you need to know. The same will be with my scenario (but I will have only less than one hundred vars).

**Comment by scrubbless** [comment]
It will be a lot of pointless variables. If specific vars are required, then it's better to single them out. Could be a needle in a haystack checking a list of all vars.

**Comment by BenRQ** [comment]
It's an easy win really in term of development, and there are a lot of interesting data in there.

A small scenario starts with 30 variables, so reaching 100 is quite fine.
And even with a hundred vars, it's still quite small amount of data.

It should look like this:

$restock:0;#BtT:1;#CotWI:1;#CotWM:1;#CotWT:1;#FAI:1;#FAM:1;#FAT:1;#MoM1EI:1;#MoM1EM:1;#MoM1ET:1;#RN:1;#SM:1;#SoA:1;#SoT:1;#MoM1E:1;#CotW:1;#FA:1;#round:1;#HeroMarieLambeau:1;#HeroHarveyWalters:1;#HeroMinhThiPhan:1;$%morale:3;#heroes:3;FantasmaEND:0;Caida:0;SotanoActivo:0;Tunel:0;Mythos7LOST:0;MythosSoft8:0;DEAD:0;MythosHard:0;$end:0;

**Comment by scrubbless** [comment]
Can you spot out the vars for a strain on reality, that might prove or disprove my point :)

**Comment by scrubbless** [comment]
But those expansion variables are very useful. It like to see some of the vars if it's not too cluttered.

**Comment by BenRQ** [comment]
I agree it's quite a lot for stress and strain, but still manageable :)

`[Vars];#RN=1;#MoM1ET=1;#MoM1EM=1;#MoM1EI=1;#SM=1;#FAM=1;#CotWM=1;#FAI=1;#CotWI=1;#FAT=1;#CotWT=1;#BtT=1;#MoM1E=1;#CotW=1;#FA=1;#round=6;$%morale=3;#HeroCarsonSinclair=1;#HeroBobJenkins=1;#HeroAmandaSharpe=1;#heroes=3;RitualBook_VAR=5;RB_Red_Selected=1;RB_Grn_Selected=1;RB_Blu_Selected=1;RB_Yel_Selected=1;RB_Yel_WIN_Loc=1;RB_Blu_WIN_Loc=2;RB_Red_WIN_Loc=3;RB_Grn_WIN_Loc=4;GEM_RedLoc=999;GEM_Red_Assigned=1;GEM_Loc3_Selected=1;GemLoc_5and6_Selected=1;GEM_Loc7_Selected=1;Gem_Grn_Assigned=1;GEM_Blu_Assigned=1;GEM_Yel_Assigned=1;GEM_BluLoc=3;GEM_GrnLoc=7;GEM_Loc6_Selected=1;GEM_YelLoc=6;Phase=1;DrLittleAway=2;Office_SecretDoor_PuzzleVAR=3;MYTHOS_DEADLY=15;VAR_BruiserLockBoxEvidence=3;VAR_DiaryEvidance=1;$mythosFlavor=1;$mythosMinor=1;A_Mythos_CustomTurn=8;MYTHOS_Visiting_Time=9;EndGameCountdown=20;REvent2_SpawnTimer=11;RM_NoRepeat=1;SM_NoRepeat=1;BtT_Tiles=1;COTW_Tiles=1;FA_Tiles=1;RN_Tiles=1;Ammi_2PlayerFlag=1;RealityTear_Ev_Req=3;Mythos_Warning1=6;InvestigatorsUpstairs=1;Dresser_Failed=1;BarSearch_Clue=1;TidBit_Orderly2=1;TidBit_DrLittle=1;TidBit_Zadoc=1;DrLittlesOffice_SpiritualBooks=1;TidBit_RealityTear=1;BathroomAlarm_Found=1;Bathroom_Entered=1;BruiserDistractUnlocked=1;CLUE_Sym_Eye=1;CLUE_Sym_Portal=1;CLUE_Sym_Nothing1=1;CLUE_Sym_Nothing2=1;RegisterTruth=1;RegisterUsed=1;BruiserTalk_About=1;$mythosLounge=1;LoungePlaced=1;REvent_OTHERA_Bug=5;ENDR_Running=1;T1_Shortcut_Triggered=1;Othera_WindowCuriosity=1;AMMI_NameKnown=1;PatientClues=1;RICHARD_Approached=1;OTHERA_Crawlies=1;GEM_Richard_Loc6=1;RICHARD_StoleKey=1;RICHARD_ThrownOut=1;RICHARD_LibraryQuestionClue=1;RICHARD_NoRep_Talk1=1;RICHARD_NoRep_Talk2=1;Library_Placed=1;Investigators_In_Library=1;OpenedStairsLobby=2;StairsENTERED=1;Bedrooms_Placed=1;PlacedSmallRoom=1;CLUE_Sym_Book=1;T1_Shortcut_EventRUN=1;ENDR_Ran_Shortcut=1;WindowRealityTear_Placed=1;A_MythosFlavour=1;A_MythosMinor=1;A_MythosLounge=1;RB_Yel_Have=1;RealityTears=4;ZADOC_DrLittle=1;ZADOC_DecidedPositive=1;ZADOC_DrLittle_2=1;BarToken_Placed=1;DiaryEvidenceAttained=1;T7_TearSpawn=2;RB_Blu_Have=1;Bedroom_RiftSpawned=1;BedroomRealityTearLocation=1;BedroomPortalLocation=1;Phase1_MythosMonster=7;Mythos_4=1;Mythos_7=1;BruiserAway_KnowTheDrill=1;OfficeOPENED=1;DrLittle_Locked=1;T4_BooksSearched=1;DrLittlesOffice_SpiritualBooksSearched=1;Experiments_EVIDANCEATTAINED=1;RB_Grn_Have=1;DrLittle_PapersBurned=1;HaveLibraryKey=1;RealityTear_FirstUse=1;LibraryRealityTearLocation=1;BathroomRealityTearLocation=1;Bathroom_RealityTear_Active=1;CLUE_Sym_TriggerBATHROOM=1;Library_RiftPlaced=1;Library_RealityTear_Active=1;T3_BookUsed=1;RB_HaveRitualBook=1;T3_PaintingUsed=1;Mythos_6=1;REvent_OTEHRA_Bug_RUN=1;ENDR_Ran_OtheraBugs=1;OTHERA_NameKnown=1;AMMI_SpokeAboutOthera=1;RICHARD_SpokeAboutLibBAN=1;RICHARD_StrangeLibraryBooks=1;Clue_Sym_TriggerLib=1;RB_Slot1_Full=1;RB_Slot2_Full=1;RB_Slot4_Full=1;RB_Blu_Loc=1;RB_Grn_Loc=4;RB_Yel_Loc=2;RealityTear_EvidenceAttained=1;EVIDENCE_TomesAttained=1;EndR_Ran_VisitingWarning=1;ENDR_Ran_BruiserVase=1;BruiserReturn_KnowTheDrill=1;;`

**Comment by scrubbless** [comment]
I made heavy use of variables haha, but that's not as long as I expected. I can clean it it too, on the end event I could set any variables that are not important to 0.

That's A Strain on Reality right?

Some handy information in there, like what combination of gems were used and where they slotted them. (If they understood the puzzle or not).

Regards it's not as long as I thought, so yeah totally suitable to export them into the spreadsheet.

**Comment by scrubbless** [comment]
Generic graphs like 

Average time taken, max time, minutes time, time by players, win loss rates.

If I can then create a drop-down list that feeds the variables, then it would be usable for all scenarios.
